blocklist-client/src/client/risk_client.rs

```diff
@@ -0,0 +1,352 @@
+    risk_reason: Option<String>,
+}
+
+#[derive(Deserialize, Debug)]
+struct RiskAssessment {
+    pub severity: RiskSeverity,
+    pub reason: String,
+}
+
+/// Register the user address with provider to run subsequent risk checks
+async fn register_address(
+    client: &Client,
+    config: &RiskAnalysisConfig,
+    address: &str,
+) -> Result<RegistrationResponse, Error> {
+    let api_url = register_address_path(&config.api_url);
+    let body = serde_json::json!({ "address": address });
+
+    debug!("Beginning registration for address: {address}");
+
+    let response = client
+        .post(&api_url)
+        .header("Token", &config.api_key)
+        .header("Content-Type", "application/json")
+        .json(&body)
+        .send()
+        .await?;
+
+    let checked_response = check_api_response(response).await?;
+    checked_response
+        .json::<RegistrationResponse>()
+        .await
+        .map_err(Error::from)
+}
+
+/// Check risk status associated with a registered address
+async fn get_risk_assessment(
+    client: &Client,
+    config: &RiskAnalysisConfig,
+    address: &str,
+) -> Result<RiskAssessment, Error> {
+    let api_url = risk_assessment_path(&config.api_url, address);
+    debug!("Beginning risk assessment for address: {address}");
+
+    let response = client
+        .get(&api_url)
+        .header("Token", &config.api_key)
+        .send()
+        .await?;
+
+    let checked_response = check_api_response(response).await?;
+    let resp = checked_response.json::<RiskResponse>().await?;
+
+    match resp.risk {
+        Some(risk_str) => {
+            let severity = match risk_str.as_str() {
+                "Low" => RiskSeverity::Low,
+                "Medium" => RiskSeverity::Medium,
+                "High" => RiskSeverity::High,
+                "Severe" => RiskSeverity::Severe,
+                _ => return Err(Error::InvalidRiskValue(risk_str)),
+            };
+            Ok(RiskAssessment {
+                severity,
+                reason: resp.risk_reason.unwrap_or_default(),
+            })
+        }
+        None => Err(Error::InvalidApiResponse),
+    }
+}
+
+/// Screen the provided address for blocklist status after registering it.
+/// Marks the address as not accepted if it is identified as high risk.
+pub async fn check_address(
+    client: &Client,
+    config: &RiskAnalysisConfig,
+    address: &str,
+) -> Result<BlocklistStatus, Error> {
+    // First, register the address
+    let register_response = register_address(client, config, address).await?;
+    debug!("Address registered: {}", register_response.address);
+
+    // If registration is successful, proceed to check the address
+    let RiskAssessment { severity, reason } = get_risk_assessment(client, config, address).await?;
+    debug!(
+        "Received risk assessment: Severity = {}, Reason = {}",
+        severity, reason
+    );
+
+    let is_severe = matches!(severity, RiskSeverity::Severe);
```

Comment:
Nit: This would be nice as a method on the `RiskSeverity` struct, so we could just do `severity.is_severe()`.